### PR TITLE
User login

### PR DESCRIPTION
**Pull request by zacisaacson**
Login functionality for all different user types
Redirect if users try to log in after already logged in
Create user roles

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-  helper_method :cart
+  helper_method :cart, :current_user
 
   def cart
     @cart ||= Cart.new(session[:cart] ||= Hash.new(0))
   end
 
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
```
Comment (grwthomps) [inline app/controllers/application_controller.rb:12]: We should discuss with Meg/Mike why the application controller gets hit before AND after the sessions controller.

File: spec/features/users/login_spec.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
+require 'rails_helper'
+
+RSpec.describe 'User login page' do
```
Comment (johnktravers) [inline spec/features/users/login_spec.rb:3]: Perhaps splitting these tests into separate files would make them more readable. Excellent testing though

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -1,30 +1,29 @@
-<h1 class="mt-4">Register</h1>
-<div class="justify-content-center">
-<%= form_tag "/users", class: "form-width" do %>
+<h1 class='mt-4'>Register</h1>
+
+<%= form_tag '/users', class: 'form-width' do %>
```
Comment (grwthomps) [inline app/views/users/new.html.erb:3]: Nice detail in changing the double quotes to single quotes

File: config/routes.rb [hunk 4]
```diff
@@ -1,43 +1,52 @@
...
-
-  get "/orders/new", to: "orders#new"
-  post "/orders", to: "orders#create"
-  get "/orders/:id", to: "orders#show"
-
-  get "/register", to: "users#new"
-  get "/profile", to: "users#show"
-  post "/users", to: "users#create"
-
+  get '/merchants', to: 'merchants#index'
```
Comment (grwthomps) [inline config/routes.rb:4]: We could organize our routes a little better

**Review by johnktravers — COMMENTED** [review]
Great work! It might be beneficial to split up the user login tests into separate files by user type

**Review by grwthomps — APPROVED** [review]

**Pull request merged**